execution/executor/src/components/apply_chunk_output.rs

```diff
@@ -436,3 +330,300 @@ impl ParsedTransactionOutput {
         (write_set, events, reconfig_events, gas_used, status)
     }
 }
+
+/// Helper class for calculating `InMemState` after a chunk or block of transactions are executed.
+///
+/// A new SMT is spawned in two situations:
+///   1. a state checkpoint is encountered.
+///   2. a transaction chunk or block ended (where `finish()` is called)
+///
+/// | ------------------------------------------ | -------------------------- |
+/// |  (updated_between_checkpoint_and_latest)   |  (updated_after_latest)    |
+/// \                                            \                            |
+///  checkpoint SMT                               latest SMT                  |
```

Comment:
Can you verify if my understanding is correct - we need the `latest SMT` only because you are still doing per txn SMT calculation - once we move the block based SMT model, you don't need `latest SMT` anymore and you can just calculate the next checkpoint from previous checkpoint and set of state updates?

Reply:
The last is generated at the end of a chunk. In the case of state sync, it might not be a checkpoint, in that case we generate it, but don't write it down.